include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -1129,6 +1129,27 @@ class ContiguousInternalMemoryDataFacade<MLD> final
     {
     }
 };
+
+template <typename Callback>
+void ForEachSourceNodes(ContiguousInternalMemoryDataFacade<MLD> const & facade,
```

Comment:
It was introduced to be overloaded for the offline CompressedDataFacade. Moving it to routing_base_mld.hpp will cause using DataFacade, but it depends with preprocessor option. Can make a separate header for this functions.